sensor/common/externalsrcs/handler.go

```diff
@@ -156,8 +156,8 @@ func normalizeNetworkList(listProto *sensor.IPNetworkList) {
 }
 
 func networkListsEqual(a, b *sensor.IPNetworkList) bool {
-	return sliceutils.ByteEqual(a.GetIpv4Networks(), b.GetIpv4Networks()) &&
-		sliceutils.ByteEqual(a.GetIpv6Networks(), b.GetIpv6Networks())
+	return bytes.Equal(a.GetIpv4Networks(), b.GetIpv4Networks()) &&
```

Comment:
kind of interesting `[]byte` and `string` are semantically the same, but `[]byte` is not `comparable`. It makes sense, but Go already special cases some things (like allowing `==` for slices when comparing to `nil`), so I'm surprised it didn't special case `==` for `[]byte`

Reply:
Well, they're not really semantically the same: a `string` is immutable whereas a `[]byte` is mutable. They are convertible into each other, but that's about it. A `[]byte` is a reference type and a `string` is not, and I think the expectation generally is that `==` on a reference type should check reference equality (or be forbidden), not referenced object equality.

Comparison `== nil` is a different story altogether; a bunch of otherwise incomparable types (`func`s etc.) are comparable against `nil`. There's a [proposal](https://github.com/golang/go/issues/53656) to add a nullable constraint for types that are comparable against `nil`, but not necessarily against each other.